### ISSUE TITLE
Implement System workflows heuristic - part 1

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#17006]
So when content is checked in against the workflow api, calling the system workflow, here is the rule for determining what workflow action gets fired.

1. Do we have a system workflow action on content type for the action being taken? If so, use it.
2. Does the content type have "System Workflow" scheme and do we have a system workflow action on the "System Workflow" scheme for the action being taken?  If so, use it
3. Do we have a system workflow action on any workflow scheme for the content type (schemes ordered by name, first one with a default action wins).  If so, use it
4. If we do not have a default action anywhere do a normal checkin and let the unassigned listener take care of moving it to the first Step

We also need a startup task that will map the actions on the "System Workflow" to their named counterparts, e.g.
```
SAVE      = "System Workflow > Save" action
PUBLISH   = "System Workflow > PUBLISH" action
UNPUBLISH = "System Workflow > UNPUBLISH" action
ARCHIVE   = "System Workflow > ARCHIVE" action
UNARCHIVE = "System Workflow > UNARCHIVE" action
DELETE    = "System Workflow > DELETE" action
```
We do not have a DESTROY action.

**Comment by fabrizzio-dotCMS** [comment]
- Verified that there's a start-up task that creates the mapped actions.
- Verified that the system first tries to get the system actions associated with the content-type. 
- Verified that the third option is the second option is attempting to grab a possible action from the system wf
- Also verified that if the system fails to find a matching workflow action the  auto-assign delegate takes care of the exceution

**Comment by bryanboza** [comment]
Fixed, tested on release-5.2 // Postgres // FF
